lxd-agent/response.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/lxc/lxd/shared/api"
+)
+
+type devLxdResponse struct {
+	content any
+	code    int
+	ctype   string
+}
+
+func errorResponse(code int, msg string) *devLxdResponse {
+	return &devLxdResponse{msg, code, "raw"}
+}
+
+func okResponse(ct any, ctype string) *devLxdResponse {
+	return &devLxdResponse{ct, http.StatusOK, ctype}
+}
+
+func smartError(err error) *devLxdResponse {
```

Comment:
As this can return non-error responses, can this be called `smartResponse` instead?